### DISCUSSION TITLE
How to enable gtk tray items (right click) menu after settings all:unset in global rules?

### DISCUSSION THREAD

**Discussion by mobsenpai** [Alexays/Waybar#3150] [Q&A] [answered]
I have this line 
*{all:unset;} in the css file, which also disables the use of gtk's right click menu in waybar's tray items. what can I do to enable that?

**Comment by ordy** [comment]
What do you mean? That should only remove the css style, not break the right click menu.

![sc_12846](https://github.com/Alexays/Waybar/assets/22659443/9dc343cb-6e96-4ad2-ada2-2c9f9b50157e)


Here is my blueman-manager tray icon when I right click it (with `all: unset`), then you can style `#tray` `#tray menu` and `#tray menu menuitem` to make it look better.

**Reply by mobsenpai** [reply]
Yeah I meant the css style. But I wanted to use the gtk's default and not configure my own. Is it not possible. I have to use all:unset bcz without it many weird default styles are added which make it tough to correctly configure waybar.

**Reply by ordy** [reply, marked as answer]
I can't think of any other way tbh. As far as I know, you either don't unset all and try to individually remove the rules that alter your theme (targeting buttons, labels, menu...). Or unset everything and theme it yourself.

If it was like css3 you could easily chain a few :not() rules on the `*` selector, but gtk css is pretty limited. Personally I've not had any luck with that approach.

**Comment by mobsenpai** [comment]
Thought so

**Comment by dglava** [comment]
You can add `all: unset;` to every selector but the ones for the tray. The disadvantage is that it's kinda "unclean", but at least it keeps the default tray CSS.

**Comment by sqrvrt** [comment]
I think you can ask the wild card to exclude tray things, though I'm not that familiar with this

EDIT: Best I could think of for now is simply `*:not(menu):not(menuitem)`, however that still requires undoing text color.
